spec/tracking_system_spec.rb

```diff
@@ -0,0 +1,252 @@
+      expect{@tracker.view_reservations_on("some date")}.must_raise ArgumentError
+    end
+  end
+
+
+
+  describe "#total_cost_of_reservation" do
+    before do
+      @tracker = TrackingSystem.new
+      attributes = {room_num: 1 ,start_time: Date.new(2018,8,1),end_time: Date.new(2018,9,1),price: 200.00}
+      @reservation = Reservation.new(attributes)
+    end
+
+    it "raises ArgumentError unless argument is an instance of Reservation" do
+      expect{@tracker.total_cost_of_reservation("hello there!")}.must_raise ArgumentError
+    end
+
+    it "returns an instance of a Float" do
+      expect(@tracker.total_cost_of_reservation(@reservation)).must_be_kind_of Float
+    end
+
+    it "returns a postive number" do
+      expect(@tracker.total_cost_of_reservation(@reservation)).must_equal 6200.00
+    end
+  end
+
+
+  describe "#view_available_rooms_on" do
+    before do
+      @tracker = TrackingSystem.new
+    end
+
+    it "raises ArgumentError if start_time is > end_time" do
+      @tracker = TrackingSystem.new
+      expect{@tracker.view_available_rooms_on(start_time: Date.new(2018,10,5),end_time:Date.new(2018,9,5))}.must_raise ArgumentError
+    end
+
+
+    it "returns an array" do
+      @tracker = TrackingSystem.new
+      @available_rooms = @tracker.view_available_rooms_on(start_time: Date.new(2018,8,1),end_time:Date.new(2018,9,5))
+      expect(@available_rooms).must_be_kind_of Array
+    end
+
+    it "returns an array containing instances of Room" do
+      @tracker = TrackingSystem.new
+      @available_rooms = @tracker.view_available_rooms_on(start_time: Date.new(2018,8,1),end_time:Date.new(2018,9,5))
+      expect(@available_rooms[0]).must_be_kind_of Room
+    end
+
+    it "raises ArgumentError if no rooms are available on these dates(aka they're in a block or reserved)" do
+      @tracker = TrackingSystem.new
+      @tracker.all_rooms.each do |room|
+        room.reserved_dates << {start_time: Date.new(2018,1,1), end_time: Date.new(2018,1,2)}
+      end
+      expect{@tracker.view_available_rooms_on(start_time: Date.new(2018,1,1),end_time:Date.new(2018,1,2))}.must_raise ArgumentError
+    end
+  end
+
+
+  describe "#add_reservation" do
+    before do
+      @tracker = TrackingSystem.new
+    end
+
+    it "returns an array of reservations" do
+      @reservations = @tracker.add_reservation(start_time: Date.new(2018,8,1), end_time: Date.new(2018,8,25), number_of_rooms:1)
+      @reservations = @tracker.add_reservation(start_time: Date.new(2018,8,1), end_time: Date.new(2018,8,25), number_of_rooms:6)
+      list = @tracker.reservations
+      expect(@reservations).must_be_kind_of Array
+    end
+
+    it "raises ArgumentError if there are not enough available-rooms on the requested date" do
+      expect{@tracker.add_reservation(start_time: Date.new(2018,1,1),end_time:Date.new(2018,1,2),number_of_rooms:350)}.must_raise ArgumentError
+    end
+
```

Comment:
You should also test that if you make 2 reservations in the same date range, they are for different rooms.

Also test that you can make a reservation on the same date as another reservation ends.